### ISSUE TITLE
prompt encoder 问题

### ISSUE THREAD

**Issue by WeiCL7777** [xypu98/CWSAM#1]
请问作者，论文模型中的 prompt encoder 接受何种类型的 prompt 输入？在论文中没有提及，希望咨询作者，谢谢！
![Uploading Snipaste_2024-01-07_11-48-59.png…]()

**Comment by xypu98** [comment]
prompt encoder 参数没有进行微调，也没有输入prompt

**Comment by xypu98** [comment]
> 请问作者，论文模型中的 prompt encoder 接受何种类型的 prompt 输入？在论文中没有提及，希望咨询作者，谢谢！
Uploading Snipaste_2024-01-07_11-48-59.png…

prompt encoder 参数没有进行微调，也没有输入prompt

**Comment by janexue001** [comment]
> > 请问作者，论文模型中的 prompt encoder 接受何种类型的 prompt 输入？在论文中没有提及，希望咨询作者，谢谢！
> > Uploading Snipaste_2024-01-07_11-48-59.png…
> 
> prompt encoder 参数没有进行微调，也没有输入prompt

如果没有输入prompt，mask decoder中也没有处理prompt信息吗？mask decoder是一个仅处理了图像编码的解码器吗？

**Comment by xypu98** [comment]
> > > 请问作者，论文模型中的 prompt encoder 接受何种类型的 prompt 输入？在论文中没有提及，希望咨询作者，谢谢！
> > > Uploading Snipaste_2024-01-07_11-48-59.png…
> > 
> > 
> > prompt encoder 参数没有进行微调，也没有输入prompt
> 
> 如果没有输入prompt，mask decoder中也没有处理prompt信息吗？mask decoder是一个仅处理了图像编码的解码器吗？

是的。如果不输入prompt，prompt encoder那部分就没有输出特征。如果输入prompt，prompt输出特征会和image encoder特征一起进入mask decoder。可以参考SAM原本架构。

**Comment by janexue001** [comment]
谢谢您的回复，我想再请教一下，我看您的论文图示里面有prompt的输入，根据您前面的几条回复，说是没有输入prompt，这有些矛盾。所以您的论文里面是没有用到prompt encoder的是吗？







-----原始邮件-----
发件人:"Xinyang Pu" ***@***.***>
发送时间:2024-06-22 15:15:06 (星期六)
收件人: xypu98/CWSAM ***@***.***>
抄送: janexue001 ***@***.***>, Comment ***@***.***>
主题: Re: [xypu98/CWSAM] prompt encoder 问题 (Issue #1)






请问作者，论文模型中的 prompt encoder 接受何种类型的 prompt 输入？在论文中没有提及，希望咨询作者，谢谢！
Uploading Snipaste_2024-01-07_11-48-59.png…

prompt encoder 参数没有进行微调，也没有输入prompt

如果没有输入prompt，mask decoder中也没有处理prompt信息吗？mask decoder是一个仅处理了图像编码的解码器吗？

是的。如果不输入prompt，prompt encoder那部分就没有输出特征。如果输入prompt，prompt输出特征会和image encoder特征一起进入mask decoder。可以参考SAM原本架构。

**Comment by zkjisj** [comment]
这个模型好像认为没有points、boxes和masks的提示，直接执行了原代码中的部分判决内容。
![1](https://github.com/xypu98/CWSAM/assets/88485154/980acf9d-a3b6-4ffe-86a0-8375dd3d0d07)
以下是SAM中的sparse_embedding和dense_embedding的生成过程，红框部分为这个模型中的执行部分：
![2](https://github.com/xypu98/CWSAM/assets/88485154/edd56e82-971a-40ad-a09b-6b1fb5ecc891)
由此看来应该是没有提示的输入。

**Comment by janexue001** [comment]
> 这个模型好像认为没有points、boxes和masks的提示，直接执行了原代码中的部分判决内容。 ![1](https://private-user-images.githubusercontent.com/88485154/342239663-980acf9d-a3b6-4ffe-86a0-8375dd3d0d07.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4OTY5MDksIm5iZiI6MTcyMTg5NjYwOSwicGF0aCI6Ii84ODQ4NTE1NC8zNDIyMzk2NjMtOTgwYWNmOWQtYTNiNi00ZmZlLTg2YTAtODM3NWRkM2QwZDA3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI1VDA4MzY0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2YTk1OWNmYjA1NmZiNGZkMTlhYWNmMDljYmM0NjlmZmY2ZGVkZDBiZWUzNzczZmZjNzY5NTZhZDUwN2Y2ZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lt5aJdvFQXElF1Rn57IHPLKfO1699jFYJDw0eLpADKg) 以下是SAM中的sparse_embedding和dense_embedding的生成过程，红框部分为这个模型中的执行部分： ![2](https://private-user-images.githubusercontent.com/88485154/342239681-edd56e82-971a-40ad-a09b-6b1fb5ecc891.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4OTY5MDksIm5iZiI6MTcyMTg5NjYwOSwicGF0aCI6Ii84ODQ4NTE1NC8zNDIyMzk2ODEtZWRkNTZlODItOTcxYS00MGFkLWEwOWItNmIxZmI1ZWNjODkxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI1VDA4MzY0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI1OTNlMmMwYzk4ZTg1NWZmY2RkZTVjYzRiNzIzNDczMzdjMDc1MjQ5ZjQ1YmVjYTBlOGNhMDQ5YTM1MmQyYTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qUlfCWZC3nT7n8CHPG-sWwuVfeu0U01lho0jfLj99Dg) 由此看来应该是没有提示的输入。

原理上是这样，可是我也没有找到prompt encoder的调用信息，好像连无提示的过程都没走
